### PR TITLE
feat: lime segmentation support

### PR DESCRIPTION
**Pull request by WolodjaZ**
# Contributing to Meteors

Thank you for contributing to meteors! Please follow the guidelines below to ensure a smooth and efficient contribution process.

## PR Checklist

- [x] **PR Title**: `"semantic tag: description"`

  - **Package**: Use the appropriate semantic tag for your PR:
    - `feat`: A new feature.
    - `fix`: A bug fix.
    - `docs`: Documentation only changes.
    - `style`: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
    - `refactor`: A code change that neither fixes a bug nor adds a feature.
    - `perf`: A code change that improves performance.
    - `test`: Adding missing tests or correcting existing tests.
    - `chore`: Changes to the build process or auxiliary tools and libraries such as documentation generation.
    - `build`: Changes that affect the build system or external dependencies.
  - **Example**: `"fix: typo in README.md"`

- [x] **Related Issue(s)**: If your PR fixes an issue, please link it in the PR description.

- [x] **PR Message**: **_Delete this entire checklist_** and replace it with:

  - **Description**: A detailed description of the change.
  - **Dependencies**: Any dependencies required for this change.

- [x] **Add Tests and Docs**: If adding a new features, please include:

  1. A test for the feature, preferably unit tests that do not rely on network access.
  2. An example notebook showing its use, located in the `examples` directory.

- [x] **Lint**: Ensure your code passes linting and all tests:

  - Code is formatted with `pre-commit` hooks. Run `pre-commit install` and `pre-commit run --all-files` to format your code.

- [x] **Tests**: Ensure all tests pass:
  - Run `pytest` in the root directory to run all tests.

## Additional Guidelines

- Import optional dependencies within functions.
- Avoid adding dependencies to `pyproject.toml` files unless required for unit tests.
- Most PRs should not modify more than one package.
- Ensure changes are backwards compatible.

## Review Process

If no one reviews your PR within a few days, please `@-mention` one of the maintainers.

### CODE REVIEW

File: src/meteors/lime.py [hunk 1]
```diff
@@ -1605,16 +1636,25 @@ def get_spectral_attributes(
                 If equals to None, the band mask is created within the function. Defaults to None.
             target (int, optional): If the model creates more than one output, it analyzes the given target.
                 Defaults to None.
-            band_names (list[str] | dict[str | tuple[str, ...], int] | None, optional): Band names. Defaults to None.
+            n_samples (int, optional): The number of samples to generate/analyze in LIME. The more the better but slower. Defaults to 10.
+            perturbations_per_eval (int, optional): The number of perturbations to evaluate at once (Simply the inner batch size).
+                Defaults to 4.
             verbose (bool, optional): Specifies whether to show progress during the attribution process. Defaults to False.
+            model_segmentation_postprocessing_for_segmentation_problem_type: (Callable[[torch.Tensor, torch.Tensor], torch.Tensor] | None):
```
Comment (Fersoil) [inline src/meteors/lime.py:1643]: `model_segmentation_postprocessing_for_segmentation_problem_type` seems a bit too long, would have changed it into something like `postprocess_segmentation_output_to_1d`

File: src/meteors/lime_base.py [hunk 2]
```diff
@@ -445,7 +463,10 @@ def attribute(
                 else:
                     curr_sample = self.perturb_func(inputs, **kwargs)
                 batch_count += 1
-                if self.perturb_interpretable_space and self.from_interp_rep_transform is not None:
+                mask_inps.append(get_mask_from_interp_rep_transform(curr_sample, **kwargs))
+                if (
+                    self.perturb_interpretable_space and self.from_interp_rep_transform is not None
+                ):  # TODO: think about this
```
Comment (Fersoil) [inline src/meteors/lime_base.py:469]: what does it mean? there should not be TODOs in the code

File: src/meteors/utils/utils.py [hunk 3]
```diff
@@ -43,3 +43,106 @@ def change_dtype_of_list(original_list: list[Any] | Any, dtype: Callable[[Any],
...
+
+def agg_segmentation_postprocessing(
+    soft_labels: bool = False,
+    classes_numb: int = 0,
+    class_axis: int = 1,
+    use_mask: bool = True,
+) -> Callable[[torch.Tensor, torch.Tensor], torch.Tensor]:  # pragma: no cover
+    """Postprocessing function for aggregating segmentation outputs.
+
+    This function takes the segmentation model output and sums the pixel scores for
```
Comment (Fersoil) [inline src/meteors/utils/utils.py:84]: this is the description of an inner function - not the generating one. I'd move (or copy) this description into the body of the inner function and write something about the generator as well

**Review by Fersoil — CHANGES_REQUESTED** [review]
very good job in general

File: src/meteors/utils/utils.py [hunk 4]
```diff
@@ -43,3 +43,106 @@ def change_dtype_of_list(original_list: list[Any] | Any, dtype: Callable[[Any],
         return [dtype(element) for element in original_list]
     else:
         return [dtype(original_list)]
+
+
+def adjust_shape(target: torch.Tensor, source: torch.Tensor) -> torch.Tensor:
+    """Adjust the shape of a source tensor to match the shape of target tensor.
```
[the remark this answers is not in the document]
Comment (WolodjaZ) [reply src/meteors/utils/utils.py]: Change was applied my lord

File: src/meteors/lime_base.py [hunk 5]
```diff
@@ -464,9 +464,7 @@ def attribute(
                     curr_sample = self.perturb_func(inputs, **kwargs)
                 batch_count += 1
                 mask_inps.append(get_mask_from_interp_rep_transform(curr_sample, **kwargs))
-                if (
-                    self.perturb_interpretable_space and self.from_interp_rep_transform is not None
-                ):  # TODO: think about this
+                if self.perturb_interpretable_space and self.from_interp_rep_transform is not None:
```
Comment (Fersoil) [inline src/meteors/lime_base.py:467]: good that TODO is completed then

**Review by Fersoil — APPROVED** [review]

**Pull request merged**